packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -215,9 +215,16 @@ class OverlayState extends State<Overlay> {
   }
 
   void _remove(OverlayEntry entry) {
-    setState(() {
-      _entries.remove(entry);
+    assert(() {
+      if (!mounted)
+        assert(_entries.remove(entry));
```

Comment:
Problem with not mutating it is that it doesn't catch double removals.

The assert branch here only runs when the object is being disposed, so in
theory it shouldn't matter whether the list is cleaned up or not.

I agree with your point though. My original thought was to just allow
removals by having setState by children be allowed during dispose, but matt
tried it and it apparently causes lots of test failures. We can look at
that again more closely though if you want.

On Tue, Apr 19, 2016, 23:29 Adam Barth <notifications@github.com> wrote:

> In packages/flutter/lib/src/widgets/overlay.dart
> <https://github.com/flutter/flutter/pull/3412#discussion_r60355591>:
>
> > @@ -215,9 +215,16 @@ class OverlayState extends State<Overlay> {
> >    }
> >
> >    void _remove(OverlayEntry entry) {
> > -    setState(() {
> > -      _entries.remove(entry);
> > +    assert(() {
> > +      if (!mounted)
> > +        assert(_entries.remove(entry));
>
> It's slightly scary to actually mutate the list differently in checked
> versus non-checked mode. Maybe we should just assert that
> _entries.contains(entry) ?
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/flutter/flutter/pull/3412/files/2d2cde94993228383636896754da5ad673597451#r60355591>
>